src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -512,19 +532,25 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
             {
                 sendTask = base.SendAsync(request, cts.Token);
             }
+            catch (TaskCanceledException e) when (TimeoutFired(timeoutState, cancellationToken))
+            {
+                HandleFinishSendAsyncCleanup(cts, disposeCts, timer);
+                ThrowTimeoutException(ExceptionDispatchInfo.Capture(e));
+                throw;
```

Comment:
Why the additional ```throw;```? Doesn't ```ThrowTimeoutException``` throw allways?

Reply:
To make the compiler happy, otherwise it would complain below about unassigned variables (sendTask and timer).

Reply:
At both call sites you're adding `throw;` after it.  Why not just change it from `ThrowTimeoutException` to `CreateTimeoutException` and then do `throw CreateTimeoutException(....)`?